.github/workflows/build-and-test.yml

```diff
@@ -75,6 +75,9 @@ jobs:
     runs-on: ubuntu-latest
     needs: [setup-environment]
     steps:
+      - run: sudo apt-get -qq update
```

Comment:
Could you add a comment stating that this is required because of the promtail receiver component? I read your comment that this is required for compiling this component. In that case, would the whole build fail if this component isn't present?

Reply:
By promtail receiver component isn't present you mean the repo doesn't have it in the receiver directory as well as all its mentions in other places like `internal/components.go`, `internal/components_test.go`, `versions.yaml`. Basically how it is now in the main branch, right?

If so the build will pass with or without installing libsystemd-dev.

Reply:
We could isolate the code that depends on libsystemd-dev on a specific source, which is used only when a specific build flag is passed. The makefile could then check whether this lib exists and set the appropriate build flag. For our CI, we should always enable this, but I wouldn't require people on Mac or Windows systems to have this library to build contrib locally.

This brings another question: would the promtail receiver work on Windows and Darwin builds?

Reply:
Promtail receiver will work on Windows and Darwin builds because journal target is built only for linux and when cgo is enabled: https://github.com/grafana/loki/blob/488e64e3ff1bbce70d856c3daed150b061c5614d/clients/pkg/promtail/targets/journal/journaltarget.go#L1-L2

journal target will be ignored by non-linux OS and the collector build will be successful

> We could isolate the code that depends on libsystemd-dev on a specific source, which is used only when a specific build flag is passed. The makefile could then check whether this lib exists and set the appropriate build flag.

[promtail depends on libsystemd-dev](https://github.com/grafana/loki/blob/488e64e3ff1bbce70d856c3daed150b061c5614d/clients/pkg/promtail/targets/journal/journaltarget.go#L13), and promtail receiver depends on promtail. Could you please reveal the idea in more detail? What do you mean by a specific flag? Do you mean to have a custom build flag like `linux-with-libsystemd-dev`? To make it work I will need to add comment `//go:build linux-with-libsystemd-dev` to all promtail receiver files, right? And to run build we will pass flag `go build -tags linux-with-libsystemd-dev` That will work. Or if Loki team is agree we can add specfic flag `linux-with-libsystemd-dev` to https://github.com/grafana/loki/blob/main/clients/pkg/promtail/targets/journal/journaltarget.go to isolate that code and exclude it from compilation in case linux without libsystemd-dev. It will allow exclude exactly journald part from promtail without excluding the whole promtail receiver

Reply:
maybe more clear approach would be:
- check OS in Makefile
- if OS == linux then install `libsystemd-dev` package 
@jpkrohling what do you think?

Reply:
my last comment could be ignored